### PR TITLE
#1449: Fix switch layouts

### PR DESCRIPTION
**Pull request by BLoe**
Make most areas in the page editor use normal field layouts with `as={SwitchButtonWidget}` as opposed to using the `layout="switch"` option. This looks better in most cases where we aren't rending an organized column of switches separate from other fields (like the share extension modal).

image

### CODE REVIEW

**Comment by BLoe** [comment]
Maybe `Collapsible` here shouldn't be under Advanced options? Not sure if it makes more sense to move it up into the regular Configuration section.

**Comment by twschiller** [comment]
> Maybe `Collapsible` here shouldn't be under Advanced options?

It's under advanced options because it's a bit tricky to add the collapse handler to the template (you need mark the content pane, and add the collapse handle)

**Comment by twschiller** [comment]
@BLoe looks like you need to update a test snapshot

File: src/utils.test.ts [hunk 1]
```diff
@@ -59,3 +64,26 @@ describe("removeUndefined", () => {
     });
   });
 });
+
+describe("camelCaseToSentenceCase()", () => {
```
Comment (twschiller) [inline src/utils.test.ts:68]: Additional test cases:
- Acronyms, e.g., "shadowDOM"
- Underscores (e.g., `in_Foo` is the UiPath naming convention, we fetch the params of a UiPath bot for showing in the page editor)

File: src/components/fields/fieldUtils.ts [hunk 2]
```diff
@@ -16,10 +16,11 @@
  */
 
 import { Schema, SchemaDefinition } from "@/core";
+import { camelCaseToSentenceCase } from "@/utils";
 
 export function fieldLabel(name: string): string {
   const parts = name.split(".");
-  return parts[parts.length - 1];
+  return camelCaseToSentenceCase(parts[parts.length - 1]);
```
Comment (twschiller) [inline src/components/fields/fieldUtils.ts:23]: @BLoe This is a nice idea in theory, but I suspect may yield confusing results in practice when we fetch schemas from other sources (e.g., UiPath bot arguments, Zapier arguments, OpenAPI schemas, etc.)

Another way to go would be to support the title tag (which is what RJSF does): http://json-schema.org/understanding-json-schema/reference/generic.html

**Review by twschiller — CHANGES_REQUESTED** [review]
@BLoe let's discuss the field label change and its impact

File: src/utils.test.ts (hunk 1 above)
Comment (BLoe) [reply src/utils.test.ts]: I've removed this util function for now

File: src/components/fields/fieldUtils.ts (hunk 2 above)
Comment (BLoe) [reply src/components/fields/fieldUtils.ts]: Sounds good, I've removed the util function for now and added some `label` props where it makes sense instead.

File: src/components/form/widgets/switchButton/SwitchButtonWidget.tsx [hunk 3]
```diff
@@ -15,26 +15,22 @@
...
-export type SwitchButtonProps = {
-  name: string;
-  label: ReactNode;
+export type SwitchButtonProps = Except<FieldProps, "onChange"> & {
   onChange: React.ChangeEventHandler<CheckBoxLike>;
-  value?: boolean;
 };
 
-const SwitchButton: React.FC<SwitchButtonProps> = ({
+const SwitchButtonWidget: React.FC<SwitchButtonProps> = ({
```
Comment (BALEHOK) [inline src/components/form/widgets/switchButton/SwitchButtonWidget.tsx:32]: Should the type be `CustomFieldWidget`?

**Review by twschiller — APPROVED** [review]

**Review by BALEHOK — CHANGES_REQUESTED** [review]

**Pull request merged**